### PR TITLE
Add DNS-01 solver using the GleSYS API

### PR DESCRIPTION
**Pull request by abjugard**
This is a solver that interacts with the GleSYS API to solve DNS-01 challenges.

### CODE REVIEW

File: providers/dns/glesys/glesys.go [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+func (d *DNSProvider) addTXTRecord(domain string, name string, value string, ttl int) (int, error) {
+	response, err := d.sendRequest("POST", "addrecord", addRecordRequest{
+		Domainname:    domain,
+		Host:          name,
+		Type:          "TXT",
+		Data:          value,
+		Ttl:           ttl,
+	})
+	if response != nil {
+		fmt.Printf("GleSYS DNS: %s\n", response.Response.Status.Text)
```
Comment (mholt) [inline providers/dns/glesys/glesys.go]: These would probably be better logged, rather than printed to stdout.

Comment (abjugard) [reply providers/dns/glesys/glesys.go]: I looked at how this was handled in other solvers, and none used the logging package so I skipped that in my initial push. 

I've now pushed an update which uses the logging package, and cleans up the logging a bit as well to fall in line with the rest of the log outputs.

I'm wondering if the logging is really necessary for these messages though?

**Comment by abjugard** [comment]
Already have the Caddy adapter ready in my fork, just felt unnecessary to create a PR incase this one was rejected.

**Comment by abjugard** [comment]
Was there anything more you wanted me to do on this PR @mholt?

File: providers/dns/glesys/glesys.go [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+// API to manage TXT records for a domain.
+type DNSProvider struct {
+	apiUser       string
+	apiKey        string
+	activeRecords map[string]int
+	inProgressMu  sync.Mutex
+}
+
+// NewDNSProvider returns a DNSProvider instance configured for GleSYS.
+// Credentials must be passed in the environment variable: GLESYS_API_KEY.
```
Comment (mholt) [inline providers/dns/glesys/glesys.go]: ... and GLESYS_API_USER, no?

Comment (abjugard) [reply providers/dns/glesys/glesys.go]: Correct, you're an eagle eyed one! Fixed in update.

**Comment by mholt** [comment]
Great - thanks. Just sync with the latest master and I can merge this.

**Comment by abjugard** [comment]
Done, ready for merge.

**Pull request merged**